### PR TITLE
AppArmor cluster upgrade test

### PR DESCRIPTION
**Pull request by timstclair**
Add a cluster upgrade test for AppArmor. I still need to test this (having some trouble with the cluster-upgrade tests), but wanted to start the review process.

/cc @dchen1107 @roberthbailey

### CODE REVIEW

**Comment by calebamiles** [comment]
Added `do-not-merge` label defensively. Please check in with release team before merging.

**cc: @ethernetdan, @kubernetes/kubernetes-release-managers**

**Comment by timstclair** [comment]
Blocked by https://github.com/kubernetes/kubernetes/issues/42382

**Comment by timstclair** [comment]
Manually verified this passes the upgrade tests on head.

File: test/e2e/upgrades/apparmor.go
Comment (krousey) [inline test/e2e/upgrades/apparmor.go]: Minor nit: This confused me at first. I though this would be actively polling during an upgrade. Maybe this should be called `verifyPodStayedUp`.

**Review by krousey — APPROVED** [review]

File: test/e2e/common/apparmor.go
Comment (krousey) [inline test/e2e/common/apparmor.go]: Actually, since these are in a different package now, can you give them more specific names?

Comment (timstclair) [inline test/e2e/common/apparmor.go]: The profile gets the namespace appended to it, which should be sufficient to identify which test it's connected to. Are you looking for more than that?

File: test/e2e/upgrades/apparmor.go
Comment (timstclair) [inline test/e2e/upgrades/apparmor.go]: Yeah, that was actually the original intent, but then I realized the `Get` call would fail while the apiserver is restarting, so decided to just check after the upgrade was complete. Is there a better way to do this?

I'll rename for now.

**Review by timstclair — COMMENTED** [review]
Thanks!

File: test/e2e/common/apparmor.go
Comment (krousey) [inline test/e2e/common/apparmor.go]: What I meant was these identifiers live in a package with a lot of other functionality. If this were an AppArmor specific package, these variable names would be fine. But `allowedPath` could just as easily refer to a REST resource that RBAC allows as it could a file path.

Comment (timstclair) [inline test/e2e/common/apparmor.go]: Ahh, good point. Done.

**Comment by krousey** [comment]
@timstclair I think my PR went in, so you should rebase and add a `Name` method.

**Comment by timstclair** [comment]
Done. Also squashed.

**Comment by timstclair** [comment]
@calebamiles, @kubernetes/kubernetes-release-managers - OK to remove `do-not-merge` and merge?

**Comment by roberthbailey** [comment]
I would like to see automated upgrade testing get checked in even though we are past the code freeze.

**Comment by timstclair** [comment]
Agreed - I was under the impression that test-only changes were allowed post-freeze.

**Comment by krousey** [comment]
ping @calebamiles

**Comment by ethernetdan** [comment]
Let's get this in

**Pull request merged**
